pkg/gatherers/common/gather_logs.go

```diff
@@ -22,6 +22,7 @@ type LogContainersFilter struct {
 	Namespace                string
 	LabelSelector            string
 	FieldSelector            string
+	PodNameRegexFilter       string
```

Comment:
Now when I am looking at this again, it seems weird to define `PodNameRegexFilter` in type called `LogContainersFilter`. Pod is something different than a container. Maybe we should re-think.

Reply:
@rluders can you remind me why we can't use the `LabelSelector` here please?

Reply:
The rule:

```json
"containers_logs": {
      "alert_name": "CephClusterWarningState",
      "pod_name": "rook-ceph-osd-.*",
      "container": "osd",
      "tail_lines": 30
 }
 ```
 
If I remember correctly, the pod that triggers the alert isn't the one that we want to grab logs from, instead, we want to collect the logs from `osd` containers specifically from `rook-ceph-osd-*` pods.

Reply:
BTW, here: https://github.com/openshift/insights-operator/pull/675#discussion_r971634618

Reply:
Thanks. Maybe I don't fully remember it, but this seems little suspicious to me. It would be good to share an example of the alert with all the labels and also to reproduce it in some cluster.

Reply:
```json
{
  "metric": {
    "__name__": "ALERTS",
    "alertname": "CephClusterWarningState",
    "alertstate": "firing",
    "container": "mgr",
    "endpoint": "http-metrics",
    "instance": "0.0.0.0:9283",
    "job": "rook-ceph-mgr",
    "managedBy": "ocs-storagecluster",
    "namespace": "openshift-storage",
    "pod": "rook-ceph-mgr-a-57dc547555-kvsc7",
    "service": "rook-ceph-mgr",
    "severity": "warning"
  },
  "value": [
    1674040588.66,
    "1"
  ]
}
```